functions/Install-DbaMaintenanceSolution.ps1

```diff
@@ -277,47 +277,55 @@ function Install-DbaMaintenanceSolution {
-                    if ($jobs) {
-                        $jobs | foreach { Remove-DbaAgentJob -SqlInstance $instance -Job $_.name }
-                    }
-                }   
-            }
-            
-            try {
-                Write-Message -Level Output -Message "Installing on server $SqlInstance, database $Database"
-				
+
+				Write-Message -Level Output -Message "Dropping objects created by Ola's Maintenance Solution"
+				$null = $db.Query($CleanupQuery)
+
+				# Remove Ola's Jobs
+				if ($InstallJobs -and $ReplaceExisting) {
+					Write-Message -Level Output -Message "Removing existing SQL Agent Jobs created by Ola's Maintenance Solution"
+					$jobs = Get-DbaAgentJob -SqlInstance $server | Where-Object Description -match "hallengren"
+					if ($jobs) {
+						$jobs | foreach { Remove-DbaAgentJob -SqlInstance $instance -Job $_.name }
+					}
+				}
+			}
+
+			try {
+				Write-Message -Level Output -Message "Installing on server $SqlInstance, database $Database"
+
 				foreach ($file in $listOfFiles) {
 					$shortFileName = Split-Path $file -Leaf
 					if ($required.Contains($shortFileName)) {
 						Write-Message -Level Output -Message "Installing $file"
-                        $sql = [IO.File]::ReadAllText($file)
-                        try {
-                            foreach ($query in ($sql -Split "\nGO\b")) {
-                                $null = $db.Query($query)
-                            }
-                        }
-                        catch {
-                            Stop-Function -Message "Could not execute $file in $Database on $instance" -ErrorRecord $_ -Target $db -Continue
-                        }
-                    }
-                }
-            }
-            catch {
-                Stop-Function -Message "Could not execute $file in $Database on $instance" -ErrorRecord $_ -Target $db -Continue
-            }
-        }
-        
-        if ((Test-Path $path)) {
-            Remove-Item -Path $temp\sql-server-maintenance-solution-master -Recurse -Force -ErrorAction SilentlyContinue
-        }
-        Write-Message -Level Output -Message "Installation complete"
-    }
-}
+						$sql = [IO.File]::ReadAllText($file)
+						try {
+							foreach ($query in ($sql -Split "\nGO\b")) {
+								$null = $db.Query($query)
+							}
+						}
+						catch {
+							Stop-Function -Message "Could not execute $file in $Database on $instance" -ErrorRecord $_ -Target $db -Continue
+						}
+					}
+				}
+			}
+			catch {
+				Stop-Function -Message "Could not execute $file in $Database on $instance" -ErrorRecord $_ -Target $db -Continue
+			}
+		}
+
+		if ((Test-Path $path)) {
+			Remove-Item -Path $temp\sql-server-maintenance-solution-master -Recurse -Force -ErrorAction SilentlyContinue
+		}
+		Write-Message -Level Output -Message "Installation complete"
+	}
+	end {
+		# Only here due to need for non-pooled connection in this command
+		try {
```

Comment:
This will only close the last server - so it should be moved up to be within the `foreach ($instance in $sqlinstance)`